src/renderers/common/Renderer.js

```diff
@@ -1701,17 +1705,24 @@ class Renderer {
 	 * @param {number} width - The width in logical pixels.
 	 * @param {number} height - The height in logical pixels.
 	 * @param {boolean} [updateStyle=true] - Whether to update the `style` attribute of the canvas or not.
+	 * @param {boolean} [updateDomElement=true] - Whether to update the underlying canvas element's pixel store.
 	 */
-	setSize( width, height, updateStyle = true ) {
+	setSize( width, height, updateStyle = true, updateDomElement = true ) {
 
 		// Renderer can't be resized while presenting in XR.
 		if ( this.xr && this.xr.isPresenting ) return;
 
 		this._width = width;
 		this._height = height;
 
-		this.domElement.width = Math.floor( width * this._pixelRatio );
-		this.domElement.height = Math.floor( height * this._pixelRatio );
+		if ( updateDomElement ) {
```

Comment:
this was the root of the performance regression. Changing the size of the dom element causes its backing store to be reallocated.

Reply:
Would it be possible to add a new `_ setXRLayerSize()` renderer method that encapsulates the logic you need for layer sizing? 

```js
_setXRLayerSize( width, height ) {

	this._width = width;
	this._height = height;

	this._forceViewPort = false;

	this.setViewport( 0, 0, width, height );

}
```